Automattic/wp-calypso
# Guided Onboarding Flow: Store additional data after survey is completed

Related to https://github.com/Automattic/dotcom-forge/issues/7015

## Proposed Changes

Stores additional data after the Guided onboarding flow (`/start/guided/`) segmentation survey is completed.

`is_dev_account` - Enable developer features for users that select `Creating a site for a client `option for Q1.
`site_intent` - Set site intent to `host-site` for users that select `Creating a site for a client` option for Q1.
`site_goals` - Store site goals
`onboarding_segment` - Store segment based on responses to Q1 and Q2

<img width="660" alt="CleanShot 2024-05-31 at 11 00 37@2x" src="https://github.com/Automattic/wp-calypso/assets/10482592/522318a6-e549-4df5-94f0-cbe79c1e60d3">

## Why are these changes being made?


* This data is stored to analyze users' site-building trends and for potential future use cases.

## Testing Instructions



* Checkout PR
* Apply patch: D150692-code
* Navigate to `/start/guided/` and follow the table above to complete the flow.
* Verify the correct `onboarding_segment` is set (values can be seen here: https://github.com/Automattic/wp-calypso/pull/91348/files#diff-41bf9077d219cf1aef332e3b18f974f9ab3e589504f441fe831b45f23ed7c554R55)
* Verify the correct `site_goals` are set ( values can be seen here: )
* Verify the correct `site_intent` is set for `Creating a site for a client` Q1 option (should be set to `host-site`).
* Check `/me/developer` to ensure the developer features option is toggled

## Pre-merge Checklist
